gix-archive/src/lib.rs

```diff
@@ -1,2 +1,62 @@
 #![deny(rust_2018_idioms)]
 #![forbid(unsafe_code)]
+
+mod archive {
+
+    use gix_hash::oid;
+    use gix_object::TreeRefIter;
+    use std::io::Write;
+    use std::path::PathBuf;
+    use std::vec::IntoIter;
+
+    #[derive(Debug, thiserror::Error)]
+    #[allow(missing_docs)]
+    #[error("{msg}")]
+    pub struct FailedToArchiveError {
+        msg: String,
+    }
+
+    /// Available formats
+    #[derive(PartialEq)]
+    pub enum Format {
+        Tar,
+        Zip,
+        TarGz,
```

Comment:
Both of the compresed-`tar` formats can probably take a parameter for the compression level as part of the backend options.